File: Week_03/Practicum/Python/PythonAdditional/Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        c1, c2 = [0 for _ in range(26)], [0 for _ in range(26)]

        for c in s:
            c1[ord(c) - ord('a')] += 1
        
        for c in t:
            c2[ord(c) - ord('a')] += 1

        for c in range(26):
            if c1[c] != c2[c]:
                return False

        return True

File: Week_02/Practicum/C++/group_1/valid_anagram.cpp
// https://leetcode.com/problems/valid-anagram

class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        // Early exit if sizes differ
        if (s.size() != t.size())
        {
            return false;
        }

        // Character count for 'a' to 'z'
        vector<int> charCount(26, 0);

        for (size_t i = 0; i < s.size(); i++)
        {
            charCount[s[i] - 'a']++;
            charCount[t[i] - 'a']--;
        }

        // Check if all counts are zero
        for (int count : charCount)
        {
            if (count != 0)
            {
                return false;
            }
        }

        return true;
    }
};